core/src/signal.rs

```diff
@@ -10,7 +10,7 @@ use std::thread;
 #[serde(tag = "signal_type")]
 pub enum Signal {
     Trace(ActionWrapper),
-    Consistency(ConsistencySignal),
+    Consistency(ConsistencySignal<String>),
```

Comment:
This, and the new From impl which goes from enum to String, is the main crux of the PR.